### PR TITLE
update lock-file logic

### PR DESCRIPTION
**Pull request by natlibfi-jonollil**
`id`: New file = 0 & old file = AlephId 
Old:
Make lock-file: /exlibris/aleph/u23_3/alephe/scratch/manage18_lock -> write `id`  there -> do job -> remove lock-file.

New:
Make lock-file: /exlibris/aleph/u23_3/alephe/scratch/manage18_lock.`id` -> do job -> remove lock-file

### CODE REVIEW

**Comment by natlibfi-jonollil** [comment]
https://github.com/NatLibFi/melinda-api-legacy/blob/5b5c7fc53245050ec48649148868536e62e2e489/AlephUtils.pm#L283 Needs to be updated

File: index.py [hunk 1]
```diff
@@ -171,32 +171,31 @@ def execute(params, payload):
...
-  file.write(id)
-  file.close()
+  file = open(lockfile_full_path,"x")
+
+def remove_lock(data, mode):
+  def get_record_id():
+    return data[0:9] if mode == 'OLD' else '0'
+  
+  id = get_record_id()
+  lockfile_full_path = LOCKFILE_PATH + '.' + id
```
Comment (natlibfi-arlehiko) [inline index.py]: This is already implemented in handle_lock. Move to outside scope and use in both functions.

File: index.py [hunk 2]
```diff
@@ -171,27 +172,22 @@ def execute(params, payload):
   p.stdin.write('exit\n')
  
   (stdout, stderr) = p.communicate()
-  remove_lock(payload, params['mode'])
+  remove_lock(record_id)
 
   return (stdout, stderr)
+  
+def get_record_id(data, mode):
+  return data[0:9] if mode == 'OLD' else '0'
```
Comment (natlibfi-arlehiko) [inline index.py]: Include this too `lockfile_full_path = LOCKFILE_PATH + '.' + id`.

Comment (natlibfi-arlehiko) [reply index.py]: The two functions should call that function (Which should be renamed) and that function should return the path to the file.

**Pull request merged**